evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -122,3 +122,52 @@ def transform(self, X, y=None):
             pd.DataFrame: Transformed X.
         """
         return super().transform(X, y)
+
+
+class SelectDtypeColumns(ColumnSelector):
+    """
+    Selects columns by specified datatype in input data.
+
+    Arguments:
+        columns (list(string), list(ww.LogicalType)): List of datatypes, used to determine which columns to select.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    name = "Select Datatype Columns Transformer"
+    hyperparameter_ranges = {}
+    needs_fitting = False
+
+    def _check_input_for_columns(self, X):
+        cols = self.parameters.get("columns") or []
```

Comment:
What are your thoughts on making this a no-op? 

If we do that, then we get selection by semantic tag for free. I feel like that would be useful for workflows that don't use the "standard" set of types. For example, a user might be interested in selecting all numeric features (including Age and AgeNullable) or all categorical features (CountryCode and Ordinal). It would also make it easy for users to select based on their own criteria if they set the tags correctly. For example, users may be interested in selecting based on whether a feature was created by another component or not.

Another argument for doing that is that `check_input_for_columns` was implemented because `X.ww[cols]` would error if the columns were not in the dataframe. However, `Xww.select` does not error out if the resulting dataframe is empty. I think it's ok if this returns an empty dataframe - maybe we can just raise a warning instead?

Reply:
Agreed. I think this could definitely be a no-op, since it does allow us to select by semantic tags automatically.

Additionally,, currently it doesn't allow for selection if you use the capital string logical types (ie "NaturalLanguage" vs "natural_language")
![image](https://user-images.githubusercontent.com/22552445/126556480-5d768375-faab-4212-be5f-a5d9024c1481.png)
I would think this should work since I can call `X.ww.select()` with both and get valid results